### PR TITLE
feature: separate CICS code as a cobol dialect

### PR DESCRIPTION
**Pull request by amanPras**
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] CICS analysis should work as it was working before, with a slight error messages changes here and there (including NATIVE runtime).

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/common/src/main/java/org/eclipse/lsp/cobol/common/utils/RangeUtils.java [hunk 1]
```diff
@@ -157,6 +157,6 @@ public Range moveByLine(Range range, int count) {
    * @return character size
    */
   public int charSize(Range range) {
-    return range.getEnd().getCharacter() - range.getStart().getCharacter() + 1;
+    return range.getEnd().getCharacter() - range.getStart().getCharacter();
```
Comment (GitMensch) [inline server/common/src/main/java/org/eclipse/lsp/cobol/common/utils/RangeUtils.java]: looks unrelated and may should be get pulled in earlier?

Comment (amanPras) [reply server/common/src/main/java/org/eclipse/lsp/cobol/common/utils/RangeUtils.java]: Thanks [GitMensch](https://github.com/GitMensch) for reviewing the PR.
Yes you are correct. I have cleaned this PR.

Comment (ap891843) [reply server/common/src/main/java/org/eclipse/lsp/cobol/common/utils/RangeUtils.java]: Thanks [GitMensch](https://github.com/GitMensch) for reviewing the PR.
Yes you are correct. I have cleaned this PR a little, for reviewing (still a draft).

**Comment by GitMensch** [comment]
So, according to the check-list: "your changes generate new warnings, you did not self-review nor adjust docs so far"?

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/analysis/EmbeddedCodeListener.java [hunk 2]
```diff
@@ -209,10 +175,6 @@ private void configureParser(Parser parser) {
    * @return a visitor
    */
   public ParseTreeVisitor<List<Node>> instanceVisitor(Position position, EmbeddedLanguage lang) {
-    if (EmbeddedLanguage.CICS == lang) {
-      return new CICSVisitor(position, programUri, extendedDocument);
-    }
-
     if (EmbeddedLanguage.SQL == lang) {
```
Comment (GitMensch) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/analysis/EmbeddedCodeListener.java:178]: Jut to get the picture: Do you plan to later move out DB2, too (giving the option to use other Dialects like Oracle later and giving the option to run the parser with less need for time/resource with not "registering" any EXEC SQL?

**Comment by amanPras** [comment]
> So, according to the check-list: "your changes generate new warnings, you did not self-review nor adjust docs so far"?

I am still reviewing this PR , to check if this generates new warning. 
Also, there are no doc changes as this is a refactoring. Idea is, the plugin would (should) behave in the same manner (for an end user).

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/analysis/EmbeddedCodeListener.java (hunk 2 above)
Comment (amanPras) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/analysis/EmbeddedCodeListener.java:178]: Yes, that's the eventual plan , to see everything as a dialect. Not sure if we want to support oracle or other flavors. @zimlu02 any idea?

File: server/engine/src/test/resources/cfast/case_execCicsReturn.result.json [hunk 3]
```diff
@@ -4,19 +4,14 @@
     "location": {
       "uri": "fake/path",
       "start": {
-        "line": 3,
-        "character": 9
+        "line": 3.0,
```
Comment (GitMensch) [inline server/engine/src/test/resources/cfast/case_execCicsReturn.result.json]: There is no use in having float types for line/character. I suggest to revert that file and (re?) adjust the "origin" to use plain "int" for those attributes instead (there's no use in a full `Integer` object either).

**Review by GitMensch — COMMENTED** [review]
Apart from the floating line/character this LGTM, but mind that since some months I don't do Java on a daily basis and I did not deep-dive into how the parsers and dialects are handled internally.

File: server/engine/src/test/resources/cfast/case_execCicsReturn.result.json (hunk 3 above)
Comment (amanPras) [reply server/engine/src/test/resources/cfast/case_execCicsReturn.result.json]: updated

**Comment by amanPras** [comment]
> > So, according to the check-list: "your changes generate new warnings, you did not self-review nor adjust docs so far"?
> 
> I am still reviewing this PR , to check if this generates new warning. Also, there are no doc changes as this is a refactoring. Idea is, the plugin would (should) behave in the same manner (for an end user).

updated checklist

File: server/engine/src/test/resources/cfast/case_execCicsReturn.result.json (hunk 3 above)
Comment (GitMensch) [reply server/engine/src/test/resources/cfast/case_execCicsReturn.result.json]: The open question is: Where did the float values came from in the first place.
Isn't there some place that uses a floating type that _should_ use plain `int`?

Comment (amanPras) [reply server/engine/src/test/resources/cfast/case_execCicsReturn.result.json]: The origin has it as int (org.eclipse.lsp4j.Position). 
To assert we use [GSON.toJson](https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/403ef773f601db0c6639a5b633aae2007f5b7c49/server/engine/src/test/java/org/eclipse/lsp/cobol/extendedapi/CFASTBuilderTest.java#L63) , which converts int to float (default behavior).

Also, I simply checked the structure of the JSON and didn't care for the float (my bad).

To conclude, this is localized to the tests we have and do not (or at least I believe so) affect the server behavior.

Comment (GitMensch) [reply server/engine/src/test/resources/cfast/case_execCicsReturn.result.json]: OK - while I see https://javadoc.io/doc/org.eclipse.lsp4j/org.eclipse.lsp4j/latest/org/eclipse/lsp4j/Position.html, where is `org.eclipse.lsp4j.Position` actually defined (it seems to be missing in https://github.com/eclipse-lsp4j/lsp4j/tree/main/org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j)?

I suggest that in another PR `setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE)` may be used https://stackoverflow.com/a/70229243/5027456

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/analysis/EmbeddedCodeListener.java (hunk 2 above)
Comment (ap891843) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/analysis/EmbeddedCodeListener.java:178]: https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/pull/2098

File: server/engine/src/test/resources/cfast/case_execCicsReturn.result.json (hunk 3 above)
Comment (amanPras) [reply server/engine/src/test/resources/cfast/case_execCicsReturn.result.json]: https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/pull/2098

**Review by ishche — APPROVED** [review]
